src/components/common/SearchBar.js

```diff
@@ -0,0 +1,70 @@
+import React, { Component } from "react";
+import { Text, View, ScrollView, Dimensions } from "react-native";
+import { SearchInput, SearchResults } from ".";
+import { APIConstants } from "../../constants/APIConstants";
+import PortFolioConstants from "../../constants/PortfolioConstants";
+import CreateStockEntry from "../manage/stock/CreateStockEntry";
+
+class SearchBar extends Component {
+  constructor(props) {
+    super(props);
+    this.onSelection = this.onSelection.bind(this);
+    this.reset = this.reset.bind(this);
+    this.state = { searchInput: "", searchResults: [], lastSelection: null };
+  }
+
+  handleSelectChange(searchInput) {
+    this.setState({ searchInput });
+    const query = searchInput;
+    let getStockDataURL = APIConstants.STOCK_LOOKUP_URL;
+    let stockDataURL = getStockDataURL(query);
+
+    let response = fetch(stockDataURL)
+      .then(response => response.json())
+      .then(responseData => {
+        this.setState({ searchResults: responseData.ResultSet.Result,lastSelection : null });
+      })
+      .done();
+  }
+
+  onSelection(companyInfo) {
+    let searchInput = "";
+    let searchResults = [];
+    let lastSelection = companyInfo;
+    this.setState({ searchInput, searchResults, lastSelection });
+  }
+
+  reset(){
+    let searchInput = "";
```

Comment:
If you do not intend to reassign value to a variable prefer `const` instead of `let`

Reply:
Also to add, const should be the first preference always since it not only prevents u from reassigning the properties by forcing a more immutable design by breaking the logic inside a function, separating out the concerns, and use more pure functions to compose the outcome.

Reply:
Does not mean that there is no usecase for let, but very feeble compared to const.